spec/controllers/setup_controller_spec.rb

```diff
@@ -0,0 +1,234 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+# rubocop:disable RSpec/AnyInstance
+RSpec.describe SetupController, type: :controller do
+  let(:user)   { create(:user)   }
+  let(:minion) { create(:minion) }
+
+  describe "GET /" do
+    it "gets redirected if not logged in" do
+      get :welcome
+      expect(response.status).to eq 302
+    end
+
+    context "HTML rendering" do
+      it "returns a 200 if logged in" do
+        sign_in user
+
+        get :welcome
+        expect(response.status).to eq 200
+      end
+
+      it "renders with HTML if no format was specified" do
+        sign_in user
+
+        get :welcome
+        expect(response["Content-Type"].include?("application/json")).to be_falsey
```

Comment:
This assertion doesn't check what the actual format is but rather what it's not. The test description says otherwise.